pythreejs/nbextension/pythreejs.js

```diff
@@ -442,6 +444,50 @@ define(["nbextensions/widgets/widgets/js/widget", "nbextensions/widgets/widgets/
     });
     register['FlyControlsView'] = FlyControlsView;
 
+    var TrackballControlsView = ThreeView.extend({
+        new_properties: function() {
+            ThreeView.prototype.new_properties.call(this);
+            this.array_properties.push('target');
+        },
+
+        render: function() {
+            var that = this;
+            return utils.resolve_promises_dict(this.model.get('controlling').views).then(function(views) {
+                // get the view that is tied to the same renderer
+                that.controlled_view = _.find(views, function(o) {
+                    return o.options.renderer_id === that.options.renderer_id
+                }, that);
+                that.obj = new THREE.TrackballControls(that.controlled_view.obj, that.options.dom);
+                that.register_object_parameters();
+                that.obj.noKeys = true; // turn off keyboard navigation
+                that.options.register_update(that.obj.update, that.obj);
+                that.obj.addEventListener('change', that.options.render_frame);
+                that.obj.addEventListener('start', that.options.start_update_loop);
+                that.obj.addEventListener('end', that.options.end_update_loop);
+                that.obj.addEventListener('end', function() { that.update_controlled(); });
+                // resize again because domElement.getBoundingClientRect() returned all zeros when it's first called
```

Comment:
And to handle browser window resizing, perhaps we should also hook into that event, like in the trackball controls example:

```window.addEventListener( 'resize', function() {that.obj.handleResize()}, false );```